core/plugin.go

```diff
@@ -122,15 +126,31 @@ type SubscribedPlugin interface {
 }
 
 type RequestedPlugin struct {
-	path      string
-	checkSum  [sha256.Size]byte
-	signature []byte
+	path       string
+	checkSum   [sha256.Size]byte
+	signature  []byte
+	autoLoaded bool
+	uri        *url.URL
+}
+
+// Checks if string is URL
+func isURL(url string) bool {
+	if !govalidator.IsURL(url) || !strings.HasPrefix(url, "http") {
+		return false
+	}
+	return true
 }
 
 // NewRequestedPlugin returns a Requested Plugin which represents the plugin path and signature
 // It takes the full path of the plugin (path), temp path (fileName), and content of the file (b) and returns a requested plugin and error
 // The argument b (content of the file) can be nil
 func NewRequestedPlugin(path, fileName string, b []byte) (*RequestedPlugin, error) {
+	// Checks if string is URL
+	if isURL(path) {
+		if uri, err := url.ParseRequestURI(path); err == nil && uri != nil {
```

Comment:
What happens when is there is a parse error? It won't return and it might cause a panic.